hazelcast/src/main/java/com/hazelcast/internal/bootstrap/TpcServerBootstrap.java

```diff
@@ -168,24 +161,47 @@ private void startNio() {
 
     // public for testing
     public AltoSocketConfig getClientSocketConfig() {
+        validateSocketConfig();
+
         if (nodeEngine.getConfig().getAdvancedNetworkConfig().isEnabled()) {
             ServerSocketEndpointConfig endpointConfig = (ServerSocketEndpointConfig) nodeEngine
                     .getConfig()
                     .getAdvancedNetworkConfig()
                     .getEndpointConfigs()
                     .get(EndpointQualifier.CLIENT);
 
-            if (endpointConfig == null) {
-                return null;
-            }
-
             return endpointConfig.getAltoSocketConfig();
         }
 
         // unified socket
         return nodeEngine.getConfig().getNetworkConfig().getAltoSocketConfig();
     }
 
+    private void validateSocketConfig() {
+        AdvancedNetworkConfig advancedNetworkConfig = nodeEngine.getConfig().getAdvancedNetworkConfig();
```

Comment:
What I typically do is to extract all the relevant fields in the constructor where possible so that we do not need to go to nodeEngine class.

Reply:
https://github.com/hazelcast/hazelcast/pull/23319/commits/137ca534f7cf2944f4fd32b889b7b3e497e81313